llvm/lib/Target/EraVM/EraVMISelLowering.cpp

```diff
@@ -1339,4 +1411,22 @@ void EraVMTargetLowering::AdjustInstrPostInstrSelection(MachineInstr &MI,
   // attribute in IntrinsicsEraVM.td for this intrinsic.
   if (MI.getOpcode() == EraVM::CTXGasLeft)
     MI.setFlag(MachineInstr::MIFlag::NoMerge);
+
+  // The overflow LT aka COND_OF hasn't reversal version actually, the GE can't
+  // be used according to the spec. We need to add early-clobber attribute to
+  // the output register to prevent the RegAlloc from allocating same register
+  // for output and input operands, otherwise we may get below case which can't
+  // be expanded correctly:
+  //
+  // out = SELrrr i1 %ov, in1, in2, COND_OF
+  //
+  // where out and in1 are assigned to same register.
+  if (EraVM::isSelect(MI) &&
+      getImmOrCImm(*EraVM::ccIterator(MI)) == EraVMCC::COND_OF) {
+    if (EraVM::out0Iterator(MI)->isReg()) {
+      EraVM::out0Iterator(MI)->setIsEarlyClobber();
```

Comment:
With this line, we will always add earlyclobber flag, but we need to check whether it is SELr variant of the select. In this case, we could use `argumentType` to do so.

Reply:
Hi @vladimirradosavljevic , do you mean something like below?
```c++
  if (EraVM::isSelect(MI) &&
      (EraVM::argumentType(EraVM::ArgumentKind::In0, MI.getOpcode()) ==
       EraVM::ArgumentType::Register) &&
      getImmOrCImm(*EraVM::ccIterator(MI)) == EraVMCC::COND_OF) {
    if (EraVM::out0Iterator(MI)->isReg()) {
      EraVM::out0Iterator(MI)->setIsEarlyClobber();
    } else if (EraVM::out1Iterator(MI)->isReg()) {
      EraVM::out1Iterator(MI)->setIsEarlyClobber();
    }
  }
```

Reply:
Got it and thanks. Code is updated now per your comments.